compose/service.py

```diff
@@ -359,7 +362,8 @@ def recreate_container(self,
         """
         log.info("Recreating %s..." % container.name)
         try:
-            container.stop()
+            stop_params = {} if timeout is None else {'timeout': int(timeout)}
+            container.stop(**stop_params)
```

Comment:
I think you should be able to do just 

```
container.stop(timeout=timeout)
```

Since you'e already casted timeout to an int, or it's None already.